python/foglamp/services/core/api/certificate_store.py

```diff
@@ -33,149 +30,193 @@
+
+    certs_dir = ''
+    if cert_filename.endswith('.pem'):
+        certs_dir = _get_certs_dir('/etc/certs/pem')
+    if cert_filename.endswith('.json'):
+        certs_dir = _get_certs_dir('/etc/certs/json')
 
-    # Get certs directory path
-    certs_dir = _get_certs_dir()
     found_files = _find_file(cert_filename, certs_dir)
     is_found = True if len(found_files) else False
     if is_found and should_overwrite is False:
         raise web.HTTPBadRequest(reason="Certificate with the same name already exists. "
                                         "To overwrite set the overwrite to 1")
 
-    if key_file:
-        key_file_data = data['key'].file
-        key_file_content = key_file_data.read()
-        key_file_path = str(certs_dir) + '/{}'.format(key_filename)
-        with open(key_file_path, 'wb') as f:
-            f.write(key_file_content)
-
+    keys_dir = _get_certs_dir('/etc/certs')
+    found_files = _find_file(key_filename, keys_dir)
+    is_found = True if len(found_files) else False
+    if is_found and should_overwrite is False:
+        raise web.HTTPBadRequest(reason="Key cert with the same name already exists. "
+                                        "To overwrite set the overwrite to 1")
     if cert_file:
         cert_file_data = data['cert'].file
         cert_file_content = cert_file_data.read()
         cert_file_path = str(certs_dir) + '/{}'.format(cert_filename)
         with open(cert_file_path, 'wb') as f:
             f.write(cert_file_content)
+    if key_file:
+        key_file_data = data['key'].file
+        key_file_content = key_file_data.read()
+        key_file_path = str(keys_dir) + '/{}'.format(key_filename)
+        with open(key_file_path, 'wb') as f:
+            f.write(key_file_content)
 
     # in order to bring this new cert usage into effect, make sure to
     # update config for category rest_api
     # and reboot
-    return web.json_response({"result": "{} and {} have been uploaded successfully"
-                             .format(key_filename, cert_filename)})
+    msg = "{} has been uploaded successfully".format(cert_filename)
+    if key_file:
+        msg = "{} and {} have been uploaded successfully".format(key_filename, cert_filename)
+    return web.json_response({"result": msg})
 
 
 async def delete_certificate(request):
     """ Delete a certificate
 
     :Example:
-          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp
+          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.pem
+          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.cert?type=cert
+          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.json?type=cert
+          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.pem?type=cert
+          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.key?type=key
```

Comment:
May be an example 

`curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.key`